### PR TITLE
Allow passing ssl config options for @keyv/postgres

### PR DESCRIPTION
**Pull request by wgwz**
**Please check if the PR fulfills these requirements**
- [x] Followed the [Contributing](https://github.com/jaredwray/keyv/blob/main/CONTRIBUTING.md) guidelines.
- [x] Tests for the changes have been added (for bug fixes/features) with 100% code coverage.
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes: https://github.com/jaredwray/keyv/issues/527

Current status:
- docker-compose service added for running postgres in ssl-only mode
- new test file added for SSL tests (expectation is that they all will fail right now)

Next steps:
- set up passing ssl options in KeyvPostgres
- feedback please!

### CODE REVIEW

**Comment by jaredwray** [comment]
@wgwz - looking good so far. Thanks for working on this!

**Comment by wgwz** [comment]
These postgres tests are passing locally, with the changes made in [211edbe](https://github.com/jaredwray/keyv/pull/529/commits/211edbe228b126a3c7781412559d9fb83b8aa29e). Though when I ran the full test suite I had some tests from etcd that were failing, it had some message related to grpc.

One thing that I tried to add but could not get working is ssl certificate validation (to copy what AWS RDS does for example, see "verify-ca"). https://www.postgresql.org/docs/current/libpq-ssl.html. In other words, I was aiming to have a test case where you have to pass some "ca" option in the SSL options.

But then I thought, it'd rather be me testing that node-postgres is doing it's job, since this implementation really is just passing the SSL options to the Pool constructor. LMK what you think. 

I did test locally with a connection to an RDS database that requires verify-ca, and it's working as expected.

**Comment by jaredwray** [comment]
@wgwz - Looks like the tests are failing to connect to the ssl instance after spinning it up with docker compose. 

```bash
    pg_ctl -D /var/lib/postgresql/data -l logfile start

waiting for server to start....2022-10-24 11:41:57.663 UTC [47] FATAL:  private key file "/var/lib/postgresql/server.key" has group or world access
2022-10-24 11:41:57.663 UTC [47] DETAIL:  File must have permissions u=rw (0600) or less if owned by the database user, or permissions u=rw,g=r (0640) or less if owned by root.
initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
pg_ctl: could not start server
Examine the log output.
2022-10-24 11:44:20.521 UTC [1] FATAL:  private key file "/var/lib/postgresql/server.key" has group or world access
2022-10-24 11:44:20.521 UTC [1] DETAIL:  File must have permissions u=rw (0600) or less if owned by the database user, or permissions u=rw,g=r (0640) or less if owned by root.
2022-10-24 11:44:20.521 UTC [1] LOG:  database system is shut down
2022-10-24 11:41:57.663 UTC [47] LOG:  database system is shut down
 stopped waiting
```
I haven't done much work at setting up postgres with SSL. Do you have a reference doc or guide that I can look at to see if I can help get this running?

**Comment by wgwz** [comment]
@jaredwray i found the postgresql guide to be pretty complete:

https://www.postgresql.org/docs/current/ssl-tcp.html

for this issue though, we just have to adjust the permissions of that key, which explains why these are passing in my local vs not in CI. maybe we can do that through the docker-compose file mounts? we could also write a small custom dockerfile.

File: packages/postgres/test/test-ssl.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		await keyv.get('foo');
+		t.fail();
+	} catch {
+		t.pass();
+	} finally {
+		endPool();
+	}
+});
+
+const options = {ssl: {rejectUnauthorized: false}};
```
Comment (jaredwray) [inline packages/postgres/test/test-ssl.js:21]: @wgwz  - is this the only option for the pool on ssl? Are there other options that need to happen for it to work? The reason I am asking is I want to build documentation on the README for others on how to setup Keyv using postgres and SSL option.

**Comment by jaredwray** [comment]
@wgwz - this is looking great. Can you answer the questions above and we can get this checked in?

**Comment by wgwz** [comment]
@jaredwray sorry bout that. somehow the replies i added are stuck in a "pending" state, idk how to fix it. here's my reply:

>is this the only option for the pool on ssl? Are there other options that need to happen for it to work? The reason I am asking is I want to build documentation on the README for others on how to setup Keyv using postgres and SSL option.

there are other options in [node-postgres](https://node-postgres.com/features/ssl). which in turn leads to any [options for a node tls socket](https://nodejs.org/api/tls.html#tls_class_tls_tlssocket). which in turn leads to any [options for createSecureContext](https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions) (it's a bit of a rabbit hole). but the most commonly used would be these i guess (as in the node-postgres docs):

```
  ssl: {
    rejectUnauthorized: false,
    ca: fs.readFileSync('/path/to/server-certificates/root.crt').toString(),
    key: fs.readFileSync('/path/to/client-key/postgresql.key').toString(),
    cert: fs.readFileSync('/path/to/client-certificates/postgresql.crt').toString(),
  },
```

`rejectUnauthorized` is a development convenience, which should probably not be recommended in production. for the codebase i'm using this in, we only need to specify the `ca` option. we're connecting to an AWS RDS postgres database with `verify-ca` enabled. here's more docs on [AWS postgres RDS ssl](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.SSL.html) in case it's helpful.

in the case for the tests we have here, we only need the `rejectUnauthorized` option. because there's a mismatch in the common name of the certificate and localhost or something like that. there's probably a way that it could be fixed, and then we could just change that to `true`. but it's not too big of an issue i think for the sake of these tests.

i'm thinking that because the underlying node-postgres seems to do the same type of deferral to the node tls primitives, it makes sense to do the same here, and not worry about testing all the dimensions too much.

**Comment by jaredwray** [comment]
@wgwz - thank you so much! We will be updating documentation and looking for a release the week of November 6th. Will merge this in between then and get it ready!

**Pull request merged**